File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/PinPatternSecretUnitTest.java
package at.tugraz.tc.cyfile.secret;

import com.andrognito.patternlockview.PatternLockView;

import junit.framework.Assert;

import org.junit.Before;
import org.junit.Test;

import java.lang.reflect.Field;
import java.util.Arrays;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;

public class PinPatternSecretUnitTest extends BaseUnitTest {

    @Before
    public void setup() throws NoSuchFieldException, IllegalAccessException {
        Field field = PatternLockView.class.getDeclaredField("sDotCount");
        field.setAccessible(true);
        field.set(null, 3);
    }

    @Test
    public void bothConstructorsShouldGiveEqualValue() {

        Secret secret = new PinPatternSecret("111222");

        Secret secret1 = new PinPatternSecret(Arrays.asList(
                PatternLockView.Dot.of(1, 1),
                PatternLockView.Dot.of(1, 2),
                PatternLockView.Dot.of(2, 2)));

        Assert.assertEquals(secret.getSecretValue(), secret1.getSecretValue());
    }

    @Test
    public void testEqualVerifier() {
        EqualsVerifier.forClass(PinPatternSecret.class)
                .suppress(Warning.ALL_FIELDS_SHOULD_BE_USED)
                .suppress(Warning.NONFINAL_FIELDS)
                .withPrefabValues(PinPatternSecret.class, new PinPatternSecret("111222"), new PinPatternSecret("001122"))
                .withPrefabValues(PatternLockView.Dot.class, PatternLockView.Dot.of(0, 0), PatternLockView.Dot.of(0, 1))
                .usingGetClass()
                .verify();
    }
}

File: CyFile-Android/app/src/test/kotlin/at/tugraz/tc/cyfile/secret/PinPatternSecretUnitTest.kt
package at.tugraz.tc.cyfile.secret

import at.tugraz.tc.cyfile.BaseUnitTest
import at.tugraz.tc.cyfile.secret.impl.PinPatternSecret
import com.andrognito.patternlockview.PatternLockView
import org.junit.Assert
import nl.jqno.equalsverifier.EqualsVerifier
import nl.jqno.equalsverifier.Warning
import org.junit.Before
import org.junit.Test
import java.util.*

class PinPatternSecretUnitTest : BaseUnitTest() {

    @Before
    @Throws(NoSuchFieldException::class, IllegalAccessException::class)
    fun setup() {
        val field = PatternLockView::class.java.getDeclaredField("sDotCount")
        field.isAccessible = true
        field.set(null, 3)
    }

    @Test
    fun bothConstructorsShouldGiveEqualValue() {

        val secret = PinPatternSecret("111222")

        val secret1 = PinPatternSecret(Arrays.asList<PatternLockView.Dot>(
                PatternLockView.Dot.of(1, 1),
                PatternLockView.Dot.of(1, 2),
                PatternLockView.Dot.of(2, 2)))

        Assert.assertEquals(secret.secretValue, secret1.secretValue)
    }

    @Test
    fun testEqualVerifier() {
        EqualsVerifier.forClass(PinPatternSecret::class.java)
                .suppress(Warning.ALL_FIELDS_SHOULD_BE_USED)
                .suppress(Warning.NONFINAL_FIELDS)
                .withPrefabValues(PinPatternSecret::class.java, PinPatternSecret("111222"), PinPatternSecret("001122"))
                .withPrefabValues(PatternLockView.Dot::class.java, PatternLockView.Dot.of(0, 0), PatternLockView.Dot.of(0, 1))
                .usingGetClass()
                .verify()
    }
}
